platforms/android/tangram/src/main/java/com/mapzen/tangram/MapController.java

```diff
@@ -423,172 +428,243 @@ public void setHttpHandler(final HttpHandler handler) {
+            camera.set(update.position);
+        } else {
+            camera = getCameraPosition();
+        }
+
+        if (update.bounds != null) {
+            double[] lngLatZoom = new double[3];
+            nativeGetEnclosingViewPosition(mapPointer,
+                    update.bounds[0].longitude, update.bounds[0].latitude,
+                    update.bounds[1].longitude, update.bounds[1].latitude,
+                    update.padding, lngLatZoom);
+
+            camera.longitude = lngLatZoom[0];
+            camera.latitude = lngLatZoom[1];
+            camera.zoom = (float)lngLatZoom[2];
+        }
+        if (update.lngLat != null) {
+            camera.longitude = update.lngLat.longitude;
+            camera.latitude = update.lngLat.latitude;
+        }
+        if (update.zoomTo != null) {
+            camera.zoom = update.zoomTo;
+        }
+        if (update.zoomBy != null) {
+            camera.zoom += update.zoomBy;
+        }
+        if (update.rotation != null) {
+            camera.rotation = update.rotation;
+        }
+        if (update.tilt != null) {
+            camera.tilt = update.tilt;
+        }
+
+        setCameraPositionEased(camera, duration, ease, cb);
     }
 
     /**
-     * Get the zoom level of the map view
-     * @return Zoom level; lower values show more area
+     * Get the cameraPosition of the map view
+     * @return The current camera position
      */
-    public float getZoom() {
-        checkPointer(mapPointer);
-        return nativeGetZoom(mapPointer);
+    @NonNull
+    public CameraPosition getCameraPosition() {
+        return getCameraPosition(new CameraPosition());
     }
 
     /**
-     * Set the rotation of the view
-     * @param rotation Counter-clockwise rotation in radians; 0 corresponds to North pointing up
+     * Get the camera position of the map view
+     * @param out CameraPosition to be reused as the output
+     * @return the current camera position of the map view
      */
-    public void setRotation(final float rotation) {
+    @NonNull
+    public CameraPosition getCameraPosition(@NonNull final CameraPosition out) {
         checkPointer(mapPointer);
-        nativeSetRotation(mapPointer, rotation);
+        final double[] pos = { 0, 0 };
+        final float[] zrt = { 0, 0, 0 };
+        nativeGetCameraPosition(mapPointer, pos, zrt);
+        out.longitude = pos[0];
+        out.latitude = pos[1];
+        out.zoom = zrt[0];
+        out.rotation = zrt[1];
+        out.tilt = zrt[2];
+        return out;
     }
 
     /**
-     * Set the rotation of the view with default easing
-     * @param rotation Counter-clockwise rotation in radians; 0 corresponds to North pointing up
-     * @param duration Time in milliseconds to ease to the given rotation
+     * Set the geographic position of the center of the map view
+     * @param position LngLat of the position to set
+     */
+     public void setPosition(@NonNull final LngLat position) {
```

Comment:
Do we need these direct setters? I'm guessing the idea here is to enable brevity for simple changes - but I suspect that changes to a single parameter like this are quite rare in actual applications!